src/pudl/metadata/classes.py

```diff
@@ -1632,3 +1633,45 @@ def to_sql(
                 check_values=check_values,
             )
         return metadata
+
+
+class CodeData(Base):
+    """
+    A collection of dictionaries representing codes for various PUDL fields.
+
+    Used to export to documentation.
+    """
+
+    code_list: List[dict]
+
+    @classmethod
+    def from_code_names(
+        cls, code_names: Iterable[str], csv_dir: String
+    ) -> "CodeData":
+        """
+        Construct a list of code dictionaries containing name, description, and file path to a CSV of the code label dataframe.
+
+        Args:
+            code_names: A list of Code PUDL identifiers, keys to entries in the CODE_METADATA dictionary.
+
+        """
+        code_list = []
+        for name in code_names:
+            if name in CODE_DESCRIPTIONS.keys():
+                description = CODE_DESCRIPTIONS[name]['description']
+                title_name = CODE_DESCRIPTIONS[name]['doc_name']
+            else:
+                description = None
+                title_name = name
+            csv_filepath = os.path.join(csv_dir, f"{name}.csv")
+            CODE_METADATA[name]["df"].to_csv(csv_filepath)
```

Comment:
Outputting the CSV as a side effect of instantiating the classes seems a little sketchy / magical. It seems like all the data should be stored within the class itself, and then the `to_rst()` method should deal with the mechanics of exporting the CSV to wherever it needs to go -- maybe have `to_rst()` automatically create a directory full of CSV files next to the RST file which is being generated, that is referenced from within the generated RST?